node/src/components/linear_chain_sync/event.rs

```diff
@@ -2,6 +2,15 @@ use crate::types::{ActivationPoint, Block, BlockHash};
 
 use std::fmt::{Debug, Display};
 
+use datasize::DataSize;
+
+#[derive(DataSize, Debug, PartialEq)]
+pub enum StopReason {
+    None,
```

Comment:
Seems like it would be more natural to remove this variant so it can't possibly be sent via the `Shutdown` event, and in the `LinearChainSync` hold it as `maybe_stop_reason: Option<StopReason>`?